webhooks-extension/endpoints/webhook.go

```diff
@@ -19,17 +19,43 @@ import (
 	"fmt"
 	logging "github.com/tektoncd/experimental/webhooks-extension/pkg/logging"
 	"net/http"
+	"strconv"
 	"strings"
+	"sync"
 
 	restful "github.com/emicklei/go-restful"
 	eventapi "github.com/knative/eventing-sources/pkg/apis/sources/v1alpha1"
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+var statusLock sync.Mutex
+var createLock sync.Mutex
```

Comment:
Thanks for adding the extra lock, but I think there has to be one lock for both of them instead of two locks 😄 Otherwise, we still have the same problem if a CREATE and DELETE operation happen at the same time.

Reply:
I agree, so I've renamed it to modifyingConfigMapLock and I reference the same lock in both the delete and create functions now (I'm happy `defer` is a thing and hoping I needn't manually manage the lock with this approach).

I added another two test cases.

One for doing lots of creations at once with multiple routines to check the ConfigMap integrity.

Another is for create and delete.

@ncskier please take another look.

Reply:
Oh, wow, that is a lot of logs 😅 I'm not sure how to limit the logs either. If it's not worth removing that `Debugf` call, then I think it's alright. I like that the tests are covering all of these sync issues 👍

Reply:
IKR, it's good to know what the webhooks are at the exact point of doing this call though.

I'd really prefer it if we could just turn off the logger and I don't want less iterations for obvious reasons (sometimes the tests will pass by chance).

Alternatively, sending stdout to /dev/null for these go tests as I've tried here. We test with `go test -v` and if I remove that verbose flag it's like you never ran the tests.